src/analysis/cauchy_equation.lean

```diff
@@ -0,0 +1,130 @@
+/-
+Copyright (c) 2022 Mantas Bakšys. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mantas Bakšys
+-/
+import measure_theory.measure.lebesgue.eq_haar
+
+/-!
+# Cauchy's Functional Equation
+
+This file contains the classical results about the Cauchy's functional equation
+`f (x + y) = f x + f y` for functions `f : ℝ → ℝ`. In this file, we prove that the solutions to this
+equation are linear up to the case when `f` is a Lebesgue measurable functions, while also deducing
+intermediate well-known variants.
+-/
+
+open add_monoid_hom measure_theory measure_theory.measure metric nnreal set
+open_locale pointwise topology
+
+section seminormed_group
+open topological_space
+variables {G H : Type*} [seminormed_add_group G] [topological_add_group G] [is_R_or_C H] {s : set G}
+
+lemma add_monoid_hom.continuous_of_bounded_nhds_zero (f : G →+ H) (hs : s ∈ 𝓝 (0 : G))
+  (hbounded : bounded (f '' s)) : continuous f :=
+begin
+  obtain ⟨δ, hδ, hUε⟩ := metric.mem_nhds_iff.mp hs,
+  obtain ⟨C, hC⟩ := (bounded_iff_subset_ball 0).1 (hbounded.mono $ image_subset f hUε),
+  refine continuous_of_continuous_at_zero _ (continuous_at_iff.2 $ λ ε (hε : _ < _), _),
+  simp only [gt_iff_lt, dist_zero_right, _root_.map_zero, exists_prop],
+  obtain ⟨n, hn⟩ := exists_nat_gt (C / ε),
+  obtain hC₀ | hC₀ := le_or_lt C 0,
+  { refine ⟨δ, hδ, λ x hxδ, _⟩,
+    rwa [eq_of_dist_eq_zero (dist_nonneg.antisymm' $ (mem_closed_ball.1 $ hC $ mem_image_of_mem f $
+      mem_ball_zero_iff.2 hxδ).trans hC₀), norm_zero] },
+  have hnpos : 0 < (n : ℝ) := (div_pos hC₀ hε).trans hn,
+  refine ⟨δ / n, div_pos hδ hnpos, λ x hxδ, _⟩,
+  have h2 : f (n • x) = n • f x := map_nsmul f _ _,
+  have hn' : (n : H) ≠ 0 := nat.cast_ne_zero.2 (by { rintro rfl, simpa using hnpos }),
+  simp_rw [nsmul_eq_mul, mul_comm (n : H), ←div_eq_iff hn'] at h2,
+  replace hxδ : ‖n • x‖ < δ,
+  { refine (norm_nsmul_le _ _).trans_lt _,
+    simpa only [norm_mul, real.norm_coe_nat, lt_div_iff hnpos, mul_comm] using hxδ },
+  rw [←h2, norm_div, is_R_or_C.norm_nat_cast, div_lt_iff' hnpos, ←mem_ball_zero_iff],
+  rw div_lt_iff hε at hn,
+  exact hC.trans (closed_ball_subset_ball hn) (mem_image_of_mem _ $ mem_ball_zero_iff.2 hxδ),
+end
+
+end seminormed_group
+
+variables {ι : Type*} [fintype ι] {s : set ℝ} {a : ℝ}
+
+local notation `ℝⁿ` := ι → ℝ
+
+lemma add_monoid_hom.measurable_of_continuous (f : ℝ →+ ℝ) (h : measurable f) : continuous f :=
```

Comment:
Name says `measurable_of_continuous` but the theorem says `measurable f -> continuous f`.